src/cargo/util/context/mod.rs

```diff
@@ -283,14 +284,14 @@ impl GlobalContext {
 
         GlobalContext {
             home_path: Filesystem::new(homedir),
-            shell: RefCell::new(shell),
+            shell: Mutex::new(shell),
             cwd,
             search_stop_path: None,
-            values: LazyCell::new(),
-            credential_values: LazyCell::new(),
+            values: Default::default(),
```

Comment:
nit: in the future, splitting out changes like this that can work without this change into a refactor commit before this change can make reviewing a lot easier.

On the topic of splitting commits (more for the future), personally, I would have split this up more.  Maybe a commit per field but at least a commit per conversion type